lrpar/src/lib/ctbuilder.rs

```diff
@@ -427,17 +432,27 @@ where
         self
     }
 
-    pub fn process_header(
+    /// Sets the path to the lexer sources this is for usage from within callbacks only, and
+    /// not used during the build process.
+    pub fn lexer_path(mut self, lexer_path: PathBuf) -> Self {
+        self.lexer_path = Some(lexer_path);
+        self
+    }
+
+    pub fn inspect(
         mut self,
         cb: Box<
             dyn for<'h, 'y> Fn(
                 &'h mut Header,
                 RecoveryKind,
                 &'y YaccGrammar<StorageT>,
+                &'y StateTable<StorageT>,
+                &'y StateGraph<StorageT>,
+                Option<PathBuf>,
```

Comment:
Adding this `PathBuf` for `lexer_path` is kind of a hack, we're accruing lots of parameters,
I'm not sure if in this case we could add the `lexer_path` to Lex's `Header` and we could give `CTParserBuilder` a copy of `CTLexerBuilder`s header (we may actually need to!)

Anyhow what i'm trying to say is we could consider putting the lexer path inside the header too instead of having a specific parameter for it.

Reply:
> what i'm trying to say is we could consider putting the lexer path inside the header too

I don't want to be too dogmatic about this, as this might be the right solution, but I tend to lean towards thinking of the path as an "external" thing rather than part of the header.